lib/runtimeSettings.js

```diff
@@ -156,6 +157,12 @@ function getSettingsFile (settings) {
         }
     }
 
+    // all current drivers add settings.rootDir and settings.userDir
+    if (settings.rootDir) {
+        const uibRoot = path.join(settings.rootDir, settings.userDir, 'storage', 'uibuilder').split(path.sep).join(path.posix.sep)
+        projectSettings.uiBuilder = `uibuilder: { uibRoot: '${uibRoot}' },`
```

Comment:
Still a little nervous here Ben. I DO KNOW people who have quotes in directory names and they are valid:

![image](https://github.com/FlowFuse/nr-launcher/assets/44235289/ff85c940-f31a-47b2-bb12-727d30fc04de)

However, I cannot get my head around one thing here regarding how this will end up on file (and my point may be entirely moot) - therefore I will pull this and try it out in a crazy directory

Reply:
@hardillb as suspected, this is an issue for times when a user has a quote character in the path.

![image](https://github.com/FlowFuse/nr-launcher/assets/44235289/26b0a7fb-f45a-459d-a51b-081f3f7f0567)

I think we can easily get past this using JSON.stringify.  I will try something locally and post a suggestion shortly.

Reply:
ok so using JSON.stringify solves the ambiguity for me.

#### Resulting settings.js file
![image](https://github.com/FlowFuse/nr-launcher/assets/44235289/2f74263b-c73c-4254-9e38-34caf569abd1)


#### Working instance
![image](https://github.com/FlowFuse/nr-launcher/assets/44235289/0e29a7af-f807-42c8-b5d5-f8b0b990d268)


```suggestion
        const uibRoot = path.join(settings.rootDir, settings.userDir, 'storage', 'uibuilder').split(path.sep).join(path.posix.sep)
        projectSettings.uibuilder = { uibRoot }
```

Only accept this suggestion if accepting the partner suggestion